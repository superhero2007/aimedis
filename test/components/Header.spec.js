import React from 'react';
import { mount } from 'enzyme';

import Header from 'components/Header';

const mockDispatch = jest.fn();

function setup() {
  const props = {
    dispatch: mockDispatch,
    router: {
      location: {
        pathname: '/',
      },
    },
    user: {
      isAuthenticated: false,
    },
  };

  return mount(<Header {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
