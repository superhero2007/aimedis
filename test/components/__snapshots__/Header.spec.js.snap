// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `"<header class=\\"app__header\\"><nav class=\\"navbar navbar-expand-lg navbar-light bg-light\\"><a href=\\"/\\" class=\\"navbar-brand\\"><div class=\\"app__logo\\"><img src=\\"\\" alt=\\"aimedis-landing-page\\"></div></a><button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\".app__header__menubar\\" aria-controls=\\"app__header__menubar\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"collapse navbar-collapse justify-content-end app__header__menubar\\"><ul class=\\"justify-content-between navbar-nav d-none d-lg-flex\\"><li class=\\"nav-item active\\"><a href=\\"/\\" class=\\"nav-link\\">Home</a></li><li class=\\"nav-item\\"><a href=\\"/product\\" class=\\"nav-link\\">Product</a></li><li class=\\"nav-item\\"><a href=\\"/blog\\" class=\\"nav-link\\">Blog</a></li><li class=\\"nav-item\\"><a href=\\"/about\\" class=\\"nav-link\\">About us</a></li><li class=\\"nav-item\\"><a href=\\"/institutions\\" class=\\"nav-link\\">Institutions</a></li><li class=\\"nav-item\\"><a href=\\"/token-sale\\" class=\\"nav-link\\">Token sale</a></li></ul><form class=\\"form-inline app__header__menubar__form\\"><button type=\\"button\\" class=\\"btn btn__secondary app__header__menubar__form__login mx-xl-3\\">LogIn</button><button class=\\"navbar-toggler app__header__menubar__form__close\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\".app__header__menubar\\" aria-controls=\\"app__header__menubar\\"><i class=\\"i-close\\"></i></button></form><div class=\\"app__header__menubar__footer d-block d-lg-none\\"><hr><ul><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/\\">Home</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/product\\">Product</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/blog\\">Blog</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/about\\">About us</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/institutions\\">Institutions</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/token-sale\\">Token sale</a></li><li class=\\"nav-item mt-5\\"><a class=\\"nav-link\\" href=\\"/contact\\">Contact us</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/career\\">Career<span class=\\"separator\\">|</span><span class=\\"hire\\">We are hiring!</span></a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/investor\\">For investors</a></li><li class=\\"nav-itemfalse\\"><a class=\\"nav-link\\" href=\\"/support\\">Support</a></li></ul></div></div></nav></header>"`;
